api/desecapi/serializers.py

```diff
@@ -36,13 +37,25 @@ def set_context(self, serializer_field):
     def __call__(self):
         return self.value
 
+    @staticmethod
+    def get_domain(context):
+        try:
+            domain = Domain.objects.get(name=context['view'].kwargs['name'], owner=context['request'].user.pk)
+        except Domain.DoesNotExist:
+            if Domain.objects.get(name=context['view'].kwargs['name']):
+                raise PermissionDenied
+            else:
+                raise serializers.ValidationError("Domain not found")
+
+        return domain
+
 
 class RRsetSerializer(serializers.ModelSerializer):
     # Disallow moving RRset to different zone
     domain = serializers.SlugRelatedField(read_only=True, slug_field='name',
                                           default=FromContext(
-                                              lambda context: Domain.objects.get(name=context['view'].kwargs['name'], owner=context['request'].user.pk)
-                                          ))
+                                              lambda context: FromContext.get_domain(context)
```

Comment:
The default value is because if you `POST` on `rrsets/example.com/` to create an RRset, you need to extract `example.com` from the URL and use it to set the `domain` field of the RRset. So, the field is read-only (the user cannot supply it), but if empty (on creation), we fill it with the required value.

How do you suggest solving the layer architecture issue? If the serializer needs to transform the domain name into a domain id, it seems unavoidable for the serializer to invoke a database query.